cmd/exec-client/main.go

```diff
@@ -0,0 +1,182 @@
+// exec-client starts the remote executor client.
+package main
+
+import (
+	"context"
+	"crypto/tls"
+	"crypto/x509"
+	"flag"
+	"fmt"
+	"io"
+	"log"
+	"os"
+
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials"
+
+	pb "github.com/mlaterman/remote-exec/pkg/proto"
+)
+
+// Args is a type declaration used to collect exec args through flags.
+//
+// i.e.: -arg hello -arg world
+//       should result in ["hello", "world"]`
+type Args []string
+
+// String prints Args as a []string
+func (a *Args) String() string {
+	return fmt.Sprint(*a)
+}
+
+// Set adds a value to Args
+func (a *Args) Set(value string) error {
+	*a = append(*a, value)
+	return nil
+}
+
+func main() {
+	// Client config flags; gather through env vars/file in the future.
+	cert := flag.String("cert", "certs/client-cert.pem", "The client's public cert.")
+	key := flag.String("key", "certs/client-key.pem", "The client's private key.")
+	cas := flag.String("cas", "certs/server-cert.pem", "The server CA.")
+	address := flag.String("address", "localhost:8334", "The server address")
+
+	// grpc arguements; should be defined through some other library in the future.
+	call := flag.String("call", "status", "The procedure call name. REQUIRED, one of [start,stop,status,output].")
+	id := flag.String("id", "", "The process ID. Required for [stop,status,output].")
+	cmd := flag.String("command", "", "The executable to run. Required for [start].")
+	var args Args
+	flag.Var(&args, "args", "A a repeatable flag to specify arguments, i.e., \"-args hello -args world\". Optional, used for [start].")
+	cpu := flag.Uint64("cpu", 0, "The CPU time in ms allocated to the process. Optional, used for [start].")
+	mem := flag.Uint64("mem", 0, "The memory quota in bytes allocated to the process. Optional, used for [start].")
+	dio := flag.Uint64("io", 0, "The disk read/write limit in bytes/s allocated to the process. Optional, used for [start].")
+
+	flag.Parse()
+	log.Println("Starting exec-client.")
+	validateFlags(*call)
```

Comment:
I think this is one of those cases where using a bit more advanced CLI parsing library (like cobra) might be justified so you don't have to reimplement repeatable args and validation yourself - majority of this file's code is that. It's up to you though.